File: BackEnd (Java)/BibliotecaBusiness/src/main/java/biblioteca/gestionMaterial/boImpl/BibliotecaBOImpl.java

package biblioteca.gestionMaterial.boImpl;

import biblioteca.gestionMaterial.bo.BibliotecaBO;
import biblioteca.gestionMaterial.dao.BibliotecaDAO;
import biblioteca.gestionMaterial.mysql.BibliotecaImpl;
import java.util.ArrayList;
import pe.edu.pucp.utilsarmy.gestion_de_material.model.Biblioteca;

public class BibliotecaBOImpl implements BibliotecaBO{
    private final BibliotecaDAO daobiblioteca;

    public BibliotecaBOImpl() {
        daobiblioteca=new BibliotecaImpl();
    }

    @Override
    public int insertar(Biblioteca objeto) throws Exception {
        return daobiblioteca.insertar(objeto);
    }

    @Override
    public int modificar(Biblioteca objeto) throws Exception {
        return daobiblioteca.modificar(objeto);
    }

    @Override
    public int eliminar(int idObjeto) throws Exception {
        return daobiblioteca.eliminar(idObjeto);
    }

    @Override
    public Biblioteca obtenerPorId(int idObjeto) throws Exception {
        return daobiblioteca.obtenerPorId(idObjeto);
    }

    @Override
    public ArrayList<Biblioteca> listarTodos() throws Exception {
        return daobiblioteca.listarTodos();
    }

    @Override
    public void validar(Biblioteca objeto) throws Exception {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }
    
}

File: FrontEnd (C#)/SoftProgBusiness/GestMaterial/BOI/BibliotecaBOImpl.cs
using SoftProgBusiness.GestMaterial.BO;
using SoftProgModel.GestMaterial;
using SoftProgPersistance.GestMaterial.DAO;
using SoftProgPersistance.GestMaterial.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.GestMaterial.BOI
{
    public class BibliotecaBOImpl : IBibliotecaBO
    {
        private BibliotecaDAO daoBiblioteca;
        public BibliotecaBOImpl()
        {
            daoBiblioteca = new BibliotecaImpl();
        }
        public int eliminar(int idObjeto)
        {
            return daoBiblioteca.eliminar(idObjeto);
        }

        public int insertar(Biblioteca objeto)
        {
            return daoBiblioteca.insertar(objeto);
        }

        public BindingList<Biblioteca> listarTodos()
        {
            return daoBiblioteca.listarTodos();
        }

        public int modificar(Biblioteca objeto)
        {
            return daoBiblioteca.modificar(objeto);
        }

        public Biblioteca obtenerPorId(int idObjeto)
        {
            return daoBiblioteca.obtenerPorId(idObjeto);
        }

        public void validar(Biblioteca objeto)
        {
            throw new NotImplementedException();
        }
    }
}
